src/lib/contracts.ts

```diff
@@ -51,13 +94,30 @@ function isValidRequirementType(
 	return requirement in requirementVersions;
 }
 
+function partition<T>(arr: T[], predicate: (t: T) => boolean) {
+	return arr.reduce(
+		({ pass, fail }, elem) => {
+			if (predicate(elem)) {
+				pass.push(elem);
+			} else {
+				fail.push(elem);
+			}
+			return { pass, fail };
+		},
+		{ pass: [] as T[], fail: [] as T[] },
+	);
```

Comment:
This can be simplified and made (marginally) more performant via
```suggestion
	const result = { pass: [] as T[], fail: [] as T[] };
	for (const elem of arr) {
		if (predicate(elem)) {
			pass.push(elem);
		} else {
			fail.push(elem);
		}
	}
	return result;
```
Although I would probably say it makes even more sense to just use the lodash partition method since lodash is already used in a bunch of places so you won't get the size benefit from removing lodash altogether for the sake of rewriting this one function (but using native versions definitely makes sense imo where they're possible, until a final push to do a full removal).

And if you wanted to code golf (don't do this) then you could do `(predicate(elem) ? pass : fail).push(elem)`

Reply:
Yeah, basically I wouldn't worry about these truly useful functions until at a minimum the functions with native equivalents have all moved to those equivalent (`map`, `forEach`, `includes`, etc)